src/quantizers.jl

```diff
@@ -0,0 +1,328 @@
+# In this file, we have the color quantisation algorithms
+
+function split_buckets(img, data, depth)
+    if length(data) == 0 return end
+
+    # Buckets are same size,
+    # means that each colors is assigned to
+    # equal number of pixels here.
+    if depth == 0
+        # this behavior is interesting of N0f8
+        # and of floats as it's messing up results
+        avg = RGB{N0f8}.(mean(map(x -> x[2], data)))
+        map(x -> img[x[1]] = avg, data)
+        return
+    end
+
+    # find range of colors and pick color which
+    # most difference in max val and min val
+    rmin, rmax = 1.0N0f8, 0.0N0f8
+    gmin, gmax = 1.0N0f8, 0.0N0f8
+    bmin, bmax = 1.0N0f8, 0.0N0f8
+    for (idx, color) in data
+        if (color.r > rmax) rmax = color.r end
+        if (color.g > gmax) gmax = color.g end
+        if (color.b > bmax) bmax = color.b end
+        if (color.r < rmin) rmin = color.r end
+        if (color.g < gmin) gmin = color.g end
+        if (color.b < bmin) bmin = color.b end
+    end
+
+    ind = findmax([rmax - rmin, gmax - gmin, bmax - bmin])[2]
+
+    # sort on basis of max range color
+    if ind == 1 sort!(data; by = c -> c[2].r)
+    elseif ind == 2 sort!(data; by = c -> c[2].g)
+    elseif ind == 3 sort!(data; by = c -> c[2].b) end
+
+    # start diving on basis of median index
+    medind = trunc(Int, (length(data) + 1) / 2)
+
+    # two separate buckets
+    split_buckets(img, data[1:medind], depth - 1)
```

Comment:
Does `@view(data[1:medind])` help the performance here?

Reply:
Allocations do reduce with @views
```
julia> img = gif_decode(get_example("fire.gif"));
julia> @time mediancutquantisation(img; numcolors=32) == 16 # with @view
  0.008306 seconds (165 allocations: 3.698 MiB)
julia> @time mediancutquantisation(img; numcolors=32) == 16 # without view
  0.008978 seconds (289 allocations: 8.234 MiB)
```